### PR TITLE
filenameCollisionCheck defaults to false

### LINKED ISSUE #467: filenameCollisionCheck should default to the value of no-package-directories
Currently filenameCollisionCheck defaults to true, even though no-package-directories defaults to false.  This is incorrect.  Only if no-package-directories is set should filenameCollisionCheck be true (by default).

### PR DESCRIPTION
**Pull request by brunobowden**
Defaults to True when translateArgs contains “—no-package-directories”

Fixes #467

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1 - PTAL

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy [hunk 1]
```diff
@@ -294,9 +294,18 @@ class J2objcConfig {
     /**
      * Makes sure that the translated filenames don't collide.
      * <p/>
-     * Recommended if you choose to use --no-package-directories.
+     * Set to True when translateArgs contains '--no-package-directories'.
+     * The flag flattens the directory structure and overwrites files with
+     * the same names.
      */
-    boolean filenameCollisionCheck = true
+    boolean filenameCollisionCheck = false
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy]: would be clearer to call this 'forceFilenameCollisionCheck' and then change the javadoc.
it is not the case that this variable should be set to true, in fact we are just treating it as if it was true when no-package-directories is present.

File: src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy [hunk 2]
```diff
@@ -294,9 +294,18 @@ class J2objcConfig {
...
+
+    // All access to filenameCollisionCheck should be done through this function
+    boolean filenameCollisionCheckIsSet() {
```
Comment (advayDev1) [inline src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcConfig.groovy]: name getFilenameCollisionCheck() if you do the comment above.  that way you are also using the gradle property naming convention (get/set prefixes), and accesses to `.filenameCollisionCheck` will automatically call this method

'isSet' is somewhat ambiguous - i thought at first this was checking if the user had the set the variable (as in not accepted a default value).

**Comment by advayDev1** [comment]
lgtm with comments

**Comment by brunobowden** [comment]
So Clang allows compiles with the directory structure from the command line but not through Xcode?

Just waiting for CI to complete and then I'll merge.

**Comment by brunobowden** [comment]
Merging...
The Xcode build now takes 18 minutes which is a long time.

**Comment by advayDev1** [comment]
Xcode actually passes every individual file to clang separately.  So I think it is xcode's 'choice' not to support directories.  Also what is taking 18 minutes?  Which task?

**Comment by brunobowden** [comment]
The recent Xcode builds take 18-20 minutes:
https://travis-ci.org/j2objc-contrib/j2objc-gradle/jobs/82959080

Adding the Guava was responsible, you can see that from the build history:
https://travis-ci.org/j2objc-contrib/j2objc-gradle/builds

**Comment by advayDev1** [comment]
OH: the continuous integration builds are 18 minutes.  Not like your dev Xcode builds for your project.

Yeah, that is true.  But I don't see how we can really exercise the plugin unless it is building multiple projects with dependencies - Guava's our best bet for our various features really.

**Comment by advayDev1** [comment]
This is the first time I'm actually comfortable that changes aren't breaking the plugin.  Unit-tests can only do so much with the amount of inter-tool interaction we have

**Comment by brunobowden** [comment]
The level of complexity in the system is considerable. It's a lot more
complex that your typical library.

**Comment by advayDev1** [comment]
Is 'it' our plugin? Or guava as the system-under-test?

Either way I think that's the whole benefit of Travis.  We don't need to serially spend time (or our local CPU cycles) waiting those 18 minutes.  You push and do other work in parallel.  Like with any presubmit, devs usually have enough pipelined work that a remote system thoroughly testing shouldn't delay work.

Also Travis has a feature such that for example if you are strictly changing only CONTRIBUTING or README, etc. add `'[skip ci]` anywhere in the commit, and Travis will auto-green the CL.

**Pull request merged**